DOC-2085 fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1) (#491)
tigergraph/server-docs: modules/data-loading/partials/load-part2-create-data-source.adoc

== Create Data Source Object

A data source object provides a standard interface for all supported data source types, so that loading jobs can be written without regard for the data source.

When you create the object, you specify its details (type, access credentials, etc.) in the form of a JSON object.
The JSON object can either be read in from a file or provided inline.
[NOTE]
====
Inline mode is required when creating data sources for TigerGraph Cloud instances.
====
In the following example, we create a data source named `s1`, and read its configuration information from a file called `ds_config.json`.

[source,gsql]
USE GRAPH ldbc_snb
CREATE DATA_SOURCE s1 = "ds_config.json" FOR GRAPH ldbc_snb

[NOTE]
====
Older versions of TigerGraph required a keyword after `DATA_SOURCE` such as `STREAM` or `KAFKA`.
====
[source,gsql]
.Inline JSON data format when creating a data source
CREATE DATA_SOURCE s1 = "{
  type: <type>,
  key: <value>
}" FOR GRAPH ldbc_snb

==== String Literals
String literals can be represented according to the following options
* Enclosed with double quote `"`.
* Enclosed with triple double quotes `"""`.
* Enclosed with triple single quotes `'''`.

> In the case of JSON that does not contain a colon `:` or a comma `,` the double quotes `"` can be omitted

Reviewer: ```suggestion
In the case of JSON that does not contain a colon `:` or a comma `,` the double quotes `"` can be omitted.
```

> 

Reviewer 2: Line 3

Is 'data source code' the intended identifier? Or should this have been 'data source object' ?

Reviewer 2: Line 7

Wrap this as a note

```
[NOTE]
====
Inline mode is required when creating data sources for TigerGraph Cloud instances.
====
```

In the following example, we create a data source named `s1`, and read its configuration information from a file called `ds_config.json`.

[source,gsql]
USE GRAPH ldbc_snb
CREATE DATA_SOURCE s1 = "ds_config.json" FOR GRAPH ldbc_snb

> Older versions of TigerGraph required a keyword after `DATA_SOURCE` such as `STREAM` or `KAFKA`.

Reviewer 2: line 15

wrap this as a note
```
[NOTE]
====
Older versions of TigerGraph required a keyword after `DATA_SOURCE` such as `STREAM` or `KAFKA`.
====
```

String literals can be enclosed with a double quote `"`, triple double quotes `"""`, or triple single quotes `'''`.

> Double quotes `"` in the JSON can be omitted if the key name does not contain a colon `:` or comma `,`.

Reviewer 2: Line 24, 25

This should be reformatted, perhaps into bullet points with a sub header

### String Literals
String literals can be represented according to the following options
- Enclosed with double quote `"`.
- Enclosed with triple double quotes `"""`.
- Enclosed with triple single quotes `'''`.
In the case of JSON that does not contain a colon `:` or a comma `,` the double quotes `"` can be omitted

[source,gsql]
.Alternate quote syntax for inline JSON data
CREATE DATA_SOURCE s1 = """{

> "type": "<type>",

Reviewer 2: line 34

Both a period `.` and `_` can be used for separation in the specified key name. Example: `first.second` or `first_second`.

  "key": "<value>"
}""" FOR GRAPH ldbc_snb

> Both a period `.` and `_` can be used for separation in the specified key name. Example: `first.second` or `first_second`.

Reviewer: ```suggestion
Either a period `.` or `_` can be used for separation in the specified key name. Example: `first.second` or `first_second`.
```
